flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/util/KinesisConfigUtil.java

```diff
@@ -215,4 +229,18 @@ private static void validateOptionalPositiveDoubleProperty(Properties config, St
 			}
 		}
 	}
+
+	private static void validateOptionalDateProperty(Properties config, String key, String message) {
+		if (config.containsKey(key)) {
+			try {
+				initTimestampDateFormat.parse(config.getProperty(key));
```

Comment:
Hey this throws an error if a double is passed but the next line throws an exception when a timestamp string is passed. @tzulitai  @tony810430